src/db/utils/couchOps.ts

```diff
@@ -0,0 +1,38 @@
+import { Cleaner, uncleaner } from 'cleaners'
+import { asMaybeNotFoundError, CouchDoc } from 'edge-server-tools'
+import { DocumentScope } from 'nano'
+
+export const saveToDB = async <T>(
+  db: DocumentScope<unknown>,
+  doc: CouchDoc<T>,
+  cleaner: Cleaner<CouchDoc<T>>
+): Promise<void> => {
+  try {
+    const cleanedDoc = uncleaner(cleaner)(doc)
+    await db.insert({
+      ...cleanedDoc.doc,
+      _id: cleanedDoc.id,
+      _rev: cleanedDoc.rev ?? undefined
+    })
```

Comment:
I'm still not so sure how this works since `insert` expect a nano document type but asCouchDoc cannot be dereferenced to nano. For example, although they are structurally similar, CouchDoc names id as 'id' rather than '_id'.

Reply:
An uncleaner undoes the effect of a cleaner. That means re-assembling the fields the way Couch likes:

```
const asCouchFoo = asCouchDoc(asObject({ foo: asString }))
const wasCouchFoo = uncleaner(asCouchFoo)

asCouchFoo({ _id: '1', _rev: '2', foo: '3' }) // Goes to the clean format
wasCouchFoo({ id: '1', rev: '2', doc: { foo: '3' } }) // Goes back to the couch format
```